discovery-infra/test_infra/tools/static_ips.py

```diff
@@ -12,50 +12,82 @@ def generate_macs(count):
+
+    static_ips.extend(_genetate_ips(workers_secondary_macs,
+                                    provisioning_cidrs, num_masters))
 
-    # set static ips lists
-    masters_static_ips = utils.create_ip_address_list(num_masters, masters_static_starting_ip)
-    masters_static_secondary_ips = utils.create_ip_address_list(num_masters, masters_static_secondary_starting_ip)
-    workers_static_ips = utils.create_ip_address_list(num_workers, workers_static_starting_ip)
-    workers_static_secondary_ips = utils.create_ip_address_list(num_workers, workers_static_secondary_starting_ip)
+    return static_ips
+
+
+def _genetate_ips(mac_addresses, cidrs, address_offset=0):
+
+    num_nodes = len(mac_addresses)
+
+    first_net = ip_network(cidrs[0])
+    ipv4_gen = _static_conf_gen(
+        num_nodes, first_net, address_offset) if first_net.version == 4 else _empty_conf_gen()
 
-    mask = str(ipaddress.IPv4Network(machine_cidr).prefixlen)
-    mask_secondary = str(ipaddress.IPv4Network(provisioning_cidr).prefixlen)
-    gw_dns = str(ipaddress.ip_address(ipaddress.IPv4Network(machine_cidr).network_address) + 1)
-    gw_dns_secondary = str(ipaddress.ip_address(ipaddress.IPv4Network(provisioning_cidr).network_address) + 1)
+    if first_net.version == 6:
+        ipv6_gen = _static_conf_gen(num_nodes, first_net, address_offset)
+    else:
+        ipv6_gen = _static_conf_gen(num_nodes, ip_network(
+            cidrs[1]), address_offset) if len(cidrs) > 1 else _empty_conf_gen()
 
     static_ips = []
-    for netdata in [(masters_macs, masters_static_ips, mask, gw_dns, num_masters),
-                    (masters_secondary_macs, masters_static_secondary_ips, mask_secondary, gw_dns_secondary, num_masters),
-                    (workers_macs, workers_static_ips, mask, gw_dns, num_workers),
-                    (workers_secondary_macs, workers_static_secondary_ips, mask_secondary, gw_dns_secondary, num_workers)]:
-        data = [{'mac': netdata[0][i], 'ip': netdata[1][i], 'mask': netdata[2], 'gateway': netdata[3], 'dns': netdata[3]} for i in range(netdata[4])]
-        static_ips = static_ips + data
+    for mac in mac_addresses:
+        ipv4_conf = next(ipv4_gen)
+        ipv6_conf = next(ipv6_gen)
+        static_ips.append({
+            'mac': mac,
+            'ip': ipv4_conf[0], 'mask': ipv4_conf[1], 'gateway': ipv4_conf[2], 'dns': ipv4_conf[3],
+            'ip_v6': ipv6_conf[0], 'mask_v6': ipv6_conf[1], 'gateway_v6': ipv6_conf[2], 'dns_v6': ipv6_conf[3]
+        })
 
     return static_ips
 
+
+def _static_conf_gen(num_nodes, network, address_offset=0):
+    starting_ip = str(ip_address(network.network_address)
+                      + 30 + address_offset)
+    ips = utils.create_ip_address_list(num_nodes, starting_ip)
+    mask = str(network.prefixlen)
+    gw_dns = str(ip_address(network.network_address) + 1)
+
+    for i in range(num_nodes):
+        yield (ips[i], mask, gw_dns, gw_dns)
```

Comment:
I don't like it so much.. The meaning of each returned value isn't clear, and afterward, you address them as indexes.. I would suggest using [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) or even better a [dataclass](https://docs.python.org/3/library/dataclasses.html).

For example
```python
from dataclasses import dataclass
from typing import Optional

@dataclass 
class StaticIPConf: 
    ip: Optional[str] = None 
    mask: Optional[str] = None
    gateway: Optional[str] = None 
    dns: Optional[str] = None 
...

In [10]: StaticIPConf()                                                             
Out[10]: StaticIPConf(ip=None, mask=None, gateway=None, dns=None)

In [12]: StaticIPConf('asd')                                                        
Out[12]: StaticIP(ip='asd', mask=None, gateway=None, dns=None)

In [13]: StaticIPConf('asd').ip                                                     
Out[13]: 'asd'

```

Then `_empty_conf_gen` won't be needed.